pkg/plugins/policies/meshtrace/plugin/xds/configurer.go

```diff
@@ -0,0 +1,206 @@
+package xds
+
+import (
+	net_url "net/url"
+	"strings"
+
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	envoy_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	envoy_trace "github.com/envoyproxy/go-control-plane/envoy/config/trace/v3"
+	envoy_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	tracingv3 "github.com/envoyproxy/go-control-plane/envoy/type/tracing/v3"
+	envoy_type "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	"github.com/pkg/errors"
+
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrace/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/proto"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	v3 "github.com/kumahq/kuma/pkg/xds/envoy/listeners/v3"
+)
+
+type Configurer struct {
+	Conf *api.MeshTrace_Conf
+
+	// Opaque string which envoy will assign to tracer collector cluster, on those
+	// which support association of named "service" tags on traces. Consumed by datadog.
+	Service          string
+	TrafficDirection envoy_core.TrafficDirection
+	Destination      string
+}
+
+var _ v3.FilterChainConfigurer = &Configurer{}
+
+func (c *Configurer) Configure(filterChain *envoy_listener.FilterChain) error {
+	if c.Conf.Backends[0] == nil {
+		return nil
+	}
+
+	backend := c.Conf.Backends[0]
+
+	return v3.UpdateHTTPConnectionManager(filterChain, func(hcm *envoy_hcm.HttpConnectionManager) error {
+		hcm.Tracing = &envoy_hcm.HttpConnectionManager_Tracing{}
+
+		if c.Conf.GetSampling() != nil {
+			if c.Conf.GetSampling().GetOverall() != nil {
+				hcm.Tracing.OverallSampling = &envoy_type.Percent{
+					Value: float64(c.Conf.Sampling.Overall.Value), // missing value will
```

Comment:
missing value will?

Reply:
Unintentionally left this part of the comment, before using wrapper around this type a missing value would be converted to 0 when an Envoy default (when value is missing) is 100%, fixed this with having the wrapper type 2 lines above. Will remove this comment.